src/main/java/com/company/ExcelWriter.java

```diff
@@ -25,13 +23,30 @@ public ExcelWriter() throws FileNotFoundException {
         spreadsheet = workbook.createSheet("Student Data");
     }
 
-    public void writeListToAnExcelFile(HashMap<Integer, List<Integer>> companies) throws IOException {
-        int rowId = 0;
-        for(int x = 0; x < listOfSOmething.size(); x++){
-            Row row = spreadsheet.createRow(rowId++);
-            Cell cell = row.createCell(0);
-            cell.setCellValue(listOfSOmething.get(x));
-       }
+    public void createHeader(){
+        List<String> headers = Arrays.asList("Company", "Q1", "Q2", "Q3");
+        row = spreadsheet.createRow(0);
+        int cellId = 0;
+        for (String obj : headers) {
+            Cell cell = row.createCell(cellId++);
+            cell.setCellValue(obj);
+        }
+    }
+
+
+    public void writeListToAnExcelFile(Map<String, List<Integer>> companies) throws IOException {
+        int rowId = 1;
+        Set<String> allKeysFromMap = companies.keySet();
+        createHeader();
+        for (String key : allKeysFromMap) {
+            row = spreadsheet.createRow(rowId++);
+            row.createCell(0).setCellValue(key);
+            int cellid = 1;
+            for (Integer obj : companies.get(key)) {
+                Cell cellOfTheCurrentRow = row.createCell(cellid++);
+                cellOfTheCurrentRow.setCellValue(obj);
+            }
+        }
        workbook.write(excelFile.getExcelFile());
```

Comment:
nie widze w tym commicie tej metody getExcel file wiec nie moge t am review przeprowadzic. ALe jak robisz write do obiektu to musisz gdzies wywolac close. Poczytaj tez o Closable interfejs taki